### PR TITLE
Added MockFile.AppendAllLine() implementation and test

### PR DESCRIPTION
**Pull request by burlistic**

### CODE REVIEW

File: TestHelpers.Tests/MockFileAppendAllLines.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            });
+
+            var file = new MockFile(fileSystem);
+
+            // Act
+            file.AppendAllLines(path, new List<string> {"line 1", "line 2", "line 3"});
+
+            // Assert
+            Assert.AreEqual(
+                "Demo text contentline 1line 2line 3",
```
Comment (tathamoddie) [inline TestHelpers.Tests/MockFileAppendAllLines.cs]: Shouldn't there be new line chars here?

    "Demo text content\r\nline 1\r\nline 2\r\nline 3"

File: TestingHelpers/MockFile.cs [hunk 2]
```diff
@@ -13,6 +13,8 @@ public class MockFile : FileBase
         readonly IMockFileDataAccessor mockFileDataAccessor;
         readonly MockPath mockPath;
 
+        public static readonly Encoding DefaultEncoding = new UTF8Encoding(false, true);
```
Comment (tathamoddie) [inline TestingHelpers/MockFile.cs]: There's already `MockFileData.DefaultEncoding`. See usage in `AppendAllText` already.

File: TestingHelpers/MockFile.cs [hunk 3]
```diff
@@ -21,7 +23,31 @@ public MockFile(IMockFileDataAccessor mockFileDataAccessor)
...
         public override void AppendAllLines(string path, IEnumerable<string> contents)
         {
-            throw new NotImplementedException("This test helper hasn't been implemented yet. They are implemented on an as-needed basis. As it seems like you need it, now would be a great time to send us a pull request over at https://github.com/tathamoddie/System.IO.Abstractions. You know, because it's open source and all.");
+            ValidateParameter(path, "path");
+
+            var concatContents = string.Empty;
+
+            foreach (var content in contents)
+            {
+                concatContents += "\r\n" + content;
```
Comment (tathamoddie) [inline TestingHelpers/MockFile.cs]: To support multiple run times, should probably be `Environment.NewLine`.

File: TestingHelpers/MockFile.cs [hunk 4]
```diff
@@ -21,7 +23,31 @@ public MockFile(IMockFileDataAccessor mockFileDataAccessor)
...
+            }
+            //var concatContents = contents.Aggregate(string.Empty, (current, content) => current + content);
```
Comment (tathamoddie) [inline TestingHelpers/MockFile.cs]: Unused code? Can we strip?

**Comment by tathamoddie** [comment]
There's a growing number of commits here. Any chance you could rebase them down to one?

**Comment by tathamoddie** [comment]
I've implemented a simpler version of this patch as 8a9c903cbcf21ca49408d967796fb27dfd5c60b4. It'll be on NuGet in the next few minutes, as 2.0.0.115 or above.

File: TestHelpers.Tests/MockFileAppendAllLinesTests.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            });
+
+            var file = new MockFile(fileSystem);
+
+            // Act
+            file.AppendAllLines(path, new List<string> {"line 1", "line 2", "line 3"});
+
+            // Assert
+            Assert.AreEqual(
+                "Demo text content\r\nline 1\r\nline 2\r\nline 3",
```
Comment (ericnewton76) [inline TestHelpers.Tests/MockFileAppendAllLinesTests.cs]: Along the lines of @tathamoddie and Environment.NewLine brought up below, I would also suggest checking for Environment.NewLine to support non-windows runtimes.

**Pull request closed without merging**